java/icicle/Pair.java

```diff
@@ -15,4 +17,27 @@ public Pair(A a, B b)
     {
         return new Pair<A,B>(a, b);
     }
+
+    public String toString()
+    {
+        return "(" + _fst.toString() + ", " + _snd.toString() + ")";
+    }
+
+    public boolean equals(Object o)
```

Comment:
@charleso would appreciate your thoughts on this. 
"icicle/Pair.java uses unchecked or unsafe operations"
is there any way to implement equals without "unsafe operations"?